SofaKernel/modules/SofaDefaultType/src/sofa/defaulttype/Vec.h

```diff
@@ -476,7 +476,7 @@ class Vec : public helper::fixed_array<real,N>
     template<class real2>
     Vec<N,real> mulscalar(real2 f) const
     {
-        static_assert(DataTypeInfo<real2>::ValidInfo && DataTypeInfo<real2>::Size==1, "");
+        //static_assert(DataTypeInfo<real2>::ValidInfo && DataTypeInfo<real2>::Size==1, "");
```

Comment:
static_assert(std::is_arithmetic< real2 >::value); ?

Reply:
Thank for the tip,
 as this is outside of the scope of the PR I let that for future work for a volunteer.

Reply:
Hey @damienmarchal ,

This PR is removing those static assertions so it does in my opinion fall into its scope... 

Either remove completely those lines by indicating in the PR description why they are no longer needed, or replace them by an alternative such as the one suggested by @fredroy .

Reply:
Removing the line yes. 
Refactoring Vec no.

Reply:
Thanks a lot @damienmarchal , it makes much more sense now.